### ISSUE TITLE
nf-core/riboseq - Module Plan

### ISSUE THREAD

**Issue by JackCurragh** [Caroline-99/riboseq_pipeline#1]
### Description of feature

Here I attempt to break down the required/desired modules for each processing stage of the workflow:

### Pre-Processing
- [X] Quality Control (FastQC)
---
- [X]  Cutadapt (Adapter Removal)
- [ ]  FASTP (Adapter Removal)
---
- [X] UMI-tools (UMI handling)
---
- [ ] Bowtie 1 (rRNA removal)
- [X] HISAT2 (rRNA removal) 
- [ ] RiboDetector
- [ ] SortMeRNA

### Alignment
- [ ] STAR (Genome Alignment)
- [X] HISAT2 (Genome Alignment) 
- [ ] Bowtie (Transcriptome Alignment)
---
- [X] Samtools (Alignment processing)
- [X] BEDtools (Alignment processing) 

### Translated ORF Detection
- [ ] RiboCode
- [ ] RiboTaper
- [ ] Ribo-TISH
- [ ] Ribotricer
- [ ] ORFik 

### Quantification 
- [ ] Salmon (Pusedoalignment + Quantification)
- [X] featurecoutns (Quantification)
- [ ] RiboCode/ORFcount

### Differential Expression 
- [ ] Riborex (DE)  (DEseq2, EdgeR, Voom)
- [ ] Anota2Seq  (DE)
- [ ] DeltaTE (DE)

Please reply here with any other packages we should consider!

**Comment by Caroline-99** [comment]
Yeah I think the module plan is very good, thank you for doing it. We could also integrate subsampling at the start of the pipeline. Ed had proposed this for my dissertation, but I couldn't do it it due to time constraints. I have now added FASTP to our pipeline and am in the process of implementing subworkflows. Once I get it working, I will update everything on github.
